engine/src/main/java/org/kigalisim/engine/state/StreamParameterization.java

```diff
@@ -162,7 +167,21 @@ public EngineNumber getRechargeIntensity() {
    * @param newValue The new recovery rate value
    */
   public void setRecoveryRate(EngineNumber newValue) {
-    recoveryRate = newValue;
+    recoveryRateRecharge = newValue;
+  }
+
+  /**
+   * Set the recovery rate percentage for a specific stage.
+   *
+   * @param newValue The new recovery rate value
+   * @param stage The recovery stage (EOL or RECHARGE)
+   */
+  public void setRecoveryRate(EngineNumber newValue, RecoveryStage stage) {
+    if (stage == RecoveryStage.EOL) {
```

Comment:
Let's switch if statements here and elsewhere similar to new style case statements.